gusto/state.py

```diff
@@ -326,14 +326,13 @@ def setup_dump(self, t, tmax, pickup=False):
             # setup output directory and check that it does not already exist
             self.dumpdir = path.join("results", self.output.dirname)
             running_tests = '--running-tests' in sys.argv or "pytest" in self.output.dirname
-            if self.mesh.comm.rank == 0 \
-               and not running_tests \
-               and path.exists(self.dumpdir) and not pickup:
-                raise IOError("results directory '%s' already exists"
-                              % self.dumpdir)
-            else:
-                if not running_tests:
-                    makedirs(self.dumpdir)
+            if self.mesh.comm.rank == 0:
+                if not running_tests and path.exists(self.dumpdir) and not pickup:
+                    raise IOError("results directory '%s' already exists"
+                                  % self.dumpdir)
+                else:
+                    if not running_tests:
+                        makedirs(self.dumpdir)
```

Comment:
This bit here fixes #190. I just left it in this PR since it was a very small change to the existing logic.

Reply:
Great - thanks! It's be good to have this in separately if the test fixes prove tricky...